perfkitbenchmarker/providers/azure/azure_disk.py

```diff
@@ -44,7 +44,33 @@
 FLAGS = flags.FLAGS
 
 DRIVE_START_LETTER = 'c'
-DISK_TYPE = {disk.STANDARD: None, disk.REMOTE_SSD: None}
+
+PREMIUM_STORAGE = 'premium-storage'
+STANDARD_DISK = 'standard-disk'
+
+DISK_TYPE = {disk.STANDARD: STANDARD_DISK,
+             disk.REMOTE_SSD: PREMIUM_STORAGE}
+
+AZURE = 'Azure'
+disk.RegisterDiskTypeMap(AZURE, DISK_TYPE)
+
+PREMIUM_STORAGE_METADATA = {
+    disk.MEDIA: disk.SSD,
+    disk.REPLICATION: disk.ZONE
+}
+
+AZURE_REPLICATION_MAP = {
+    azure_network.LRS: disk.ZONE,
+    azure_network.ZRS: disk.REGION,
+    # Deliberately omitting PLRS, because that is handled by
+    # PREMIUM_STORAGE_METADATA, and (RA)GRS, because those are
+    # asynchronously replicated.
+}
+
+LOCAL_SSD_PREFIXES = {
```

Comment:
References for Azure machine types:
https://azure.microsoft.com/en-us/documentation/articles/virtual-machines-size-specs/
https://azure.microsoft.com/en-us/pricing/details/virtual-machines/#Linux
All D, DS, D v2, G, and GS machines have local SSD. Only the Basic A and Standard A series have HDD..